src/GraphCoarsen.cpp

```diff
@@ -456,6 +600,43 @@ Vector GraphCoarsen::MakeOneNegOne(int size, int split) const
     return vect;
 }
 
+Vector GraphCoarsen::MakeOneNegOne(const Vector& constant, int split) const
+{
+    assert(split >= 0);
+
+    int size = constant.size();
+
+    Vector vect(size);
+
+    double v1_sum = 0.0;
+    double v2_sum = 0.0;
+
+    for (int i = 0; i < split; ++i)
+    {
+        v1_sum += constant[i] * constant[i];
+    }
+
+    for (int i = split; i < size; ++i)
+    {
+        v2_sum += constant[i] * constant[i];
+    }
+
+    double c1 = 1.0 / split;
+    double c2 = -c1 * (v1_sum / v2_sum);
+
+    for (int i = 0; i < split; ++i)
+    {
+        vect[i] = c1 * constant[i];
+    }
```

Comment:
Assuming we agree to set `c1 = 1.0`, we actually dont need `c1` at all in this function.